pkg/boblog/progress.go

```diff
@@ -0,0 +1,138 @@
+package boblog
```

Comment:
This should not be part of the boblog package. Please create a extra package for it. e.g. `progress`